File: bitstuffing.py
def stuffing():
    frame=list(input("Enter the data of frame to stuff\n"))
    final=[]
    count=0
    for i in range(len(frame)):
        if(frame[i]=='1'):
            final.append(frame[i])
            count+=1
            if(count==5):
                final.append('0')
                count=0
        else:
            final.append(frame[i])
            count=0
    strl=''.join(final)
    print("\nAfter Bit Stuffing")
    print(strl)

def destuffing():
    frame=list(input("Enter the data of frame to destuff\n"))
    final=[]
    count=0
    for i in range(len(frame)):
        if(count==5):
            count=0
            continue
        elif(frame[i]=='1'):
            final.append(frame[i])
            count+=1
        else:
            final.append(frame[i])
            count=0
    strl=''.join(final)
    print("\nAfter Bit Destuffing")
    print(strl)

while True:
    n=input("\n\nEnter 1 to Stuff bits in data\nEnter 2 to Destuff bits in data\nEnter 0 to quit\n")
    if(n=='1'):
        stuffing()
    elif (n=='2'):
        destuffing()
    elif (n=='0'):
        break
    else:
        print("Invalid option")

File: bitstuffing.c
#include<stdio.h>
#include<string.h>
const int MAX = 1500;

void stuffing(){
    char arr[MAX];
    char farr[MAX];
    int count=0, y=0;
    printf("Enter the data packet for stuffing\n");
    scanf("%s", arr);
    int n=strlen(arr);
    for(int x=0;x<n;x++){
        if(arr[x]=='1'){
            farr[y]=arr[x];
            y++;
            count++;
            if(count==5){
                farr[y]='0';
                y++;
                count=0;
            }
        }
        else {
            farr[y]=arr[x];
            y++;
            count=0;
        }
    }
    printf("Data After bit stuffing:\n");
    printf("\n");
    for(int x=0;x<strlen(farr);x++){
        printf("%c", farr[x]);
    }
    printf("\n");
}

void destuffing() {
    char arr[MAX];
    char farr[MAX];
    int count=0, y=0;
    printf("Enter the data packet for Destuffing\n");
    scanf("%s", arr);
    int n=strlen(arr);
    for(int x=0;x<n;x++){
        if(count==5){
            count=0;
            continue;
        }
        else if(arr[x]=='1'){
            farr[y]=arr[x];
            y++;
            count++;
        }
        else
        {
            farr[y]=arr[x];
            y++;
            count=0;
        }   
    }
    printf("Data After bit Destuffing:\n");
    printf("\n");
    for(int x=0;x<strlen(farr);x++){
        printf("%c", farr[x]);
    }
    printf("\n");   
}

int main(){
    while (1==1)
    {
        int c;
        printf("\nEnter 1 to BitStuff\nEnter 2 to BitDeStuff\nEnter 0 to quit\n");
        scanf("%d", &c);
        if(c==1){
            stuffing();
        }
        else if(c==2){
            destuffing();
        }
        else if(c==0){
            break;
        }
        else {
            printf("Invalid Input\n");
        }
    }
}